examples/objectives_examples.ipynb

```diff
@@ -0,0 +1,384 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Objectives in PyJobShop\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://githubtocolab.com/PyJobShop/PyJobShop/blob/main/examples/objectives_examples.ipynb)\n",
+    "\n",
+    "> If you're using this notebook in Google Colab, be sure to install PyJobShop first by executing ```pip install pyjobshop``` in a cell."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook provides an overview of the different objectives in PyJobShop via simple scheduling problems, specifically:\n",
+    "\n",
+    "- How to set different objective weights using PyJobShop's `Model` interface, \n",
+    "- How to set the job weights with `Model`, \n",
+    "- Giving an overview of the different objectives in PyJobShop,\n",
+    "- How the objective value of a schedule is calculated in PyJobShop."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For illustration, let's add a single machine and some data to a PyJobShop `Model`. We add two jobs, each consisting of one task. The jobs' due dates differ. Furthermore, switching from the second task to the first task has a setup time of 10 time units. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-12-04T19:04:47.559191Z",
+     "start_time": "2024-12-04T19:04:46.683328Z"
+    }
+   },
+   "source": [
+    "from pyjobshop import Model\n",
+    "\n",
+    "model = Model()\n",
+    "machine = model.add_machine()\n",
+    "\n",
+    "jobs = [\n",
+    "    model.add_job(due_date=12),\n",
+    "    model.add_job(due_date=1),\n",
+    "]\n",
+    "tasks = [model.add_task(job=job) for job in jobs]\n",
+    "modes = [model.add_mode(task, machine, duration=1) for task in tasks]\n",
+    "model.add_setup_time(machine, tasks[1], tasks[0], 10)"
+   ],
+   "outputs": [],
+   "execution_count": 1
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's solve this model and inspect the results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-12-04T19:04:47.621871Z",
+     "start_time": "2024-12-04T19:04:47.563186Z"
+    }
+   },
+   "source": [
+    "result = model.solve(display=False)\n",
+    "print(\"Solver status =\", result.status)\n",
+    "print(\"Objective value =\", result.objective)\n",
+    "for idx, task in enumerate(result.best.tasks):\n",
+    "    print(f\"Start time Task {idx} =\", task.start)"
```

Comment:
Nit (this is super nit): It'd be nice to have an extra newline here between the result and print statements. They concern separate steps, so adding a newline adds a visual clue that we are moving over from one thing (solving and getting the result) to the next one (printing the result).

```python
result = model.solve(display=False)

print("Solver status =", result.status)
print("Objective value =", result.objective)
for idx, task in enumerate(result.best.tasks):
    print(f"Start time Task {idx} =", task.start)
```

This also holds for the other cells where the model is solved & printed.

Reply:
This is purely stylistic and there's not a strict rule all the time, but in this particular case it's definitely better to have this an extra line.